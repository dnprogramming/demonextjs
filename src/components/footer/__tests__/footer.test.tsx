import { render, screen } from "@testing-library/react";
import Footer from "../footer";
import React from "react";

describe("Footer Component", () => {
    it("renders footer with correct title", () => {
        render(<Footer />);
        const titleElement = screen.getByText(/Demo Next.JS/i);
        expect(titleElement).toBeInTheDocument();
    });

    it("renders footer with correct description", () => {
        render(<Footer />);
        const descriptionElement = screen.getByText(
            /All rights reserved./i
        );
        expect(descriptionElement).toBeInTheDocument();
    });

    it("renders footer component unchanged", () => {
        const { container } = render(<Footer />);
        expect(container).toMatchSnapshot();
    });
});