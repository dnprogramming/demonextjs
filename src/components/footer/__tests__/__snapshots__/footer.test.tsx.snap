// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Footer Component renders footer component unchanged 1`] = `
<div>
  <div
    id="footer"
  >
    <p>
      Â©
      2025
       Demo Next.JS.
      <br />
      All rights reserved.
    </p>
  </div>
</div>
`;
